lib/PhasarLLVM/TypeHierarchy/LLVMVFTable.cpp

```diff
@@ -48,4 +50,33 @@ nlohmann::json LLVMVFTable::getAsJson() const {
   return J;
 }
 
+std::vector<const llvm::Function *>
+LLVMVFTable::getVFVectorFromIRVTable(const llvm::ConstantStruct *VT) {
+  std::vector<const llvm::Function *> VFS;
+  for (const auto &Op : VT->operands()) {
+    if (auto *CA = llvm::dyn_cast<llvm::ConstantArray>(Op)) {
+      for (auto It = CA->operands().begin() + 2; It != CA->operands().end();
+           ++It) {
+        auto &COp = *It;
+        if (auto *CE = llvm::dyn_cast<llvm::ConstantExpr>(COp)) {
+          if (auto *BC = llvm::dyn_cast<llvm::BitCastOperator>(CE)) {
+            // if the entry is a GlobalAlias, get its Aliasee
+            auto *ENTRY = BC->getOperand(0);
+            while (auto *GA = llvm::dyn_cast<llvm::GlobalAlias>(ENTRY)) {
+              ENTRY = GA->getAliasee();
+            }
+            auto *F = llvm::dyn_cast<llvm::Function>(ENTRY);
+            VFS.push_back(F);
+          } else {
+            VFS.push_back(nullptr);
```

Comment:
What is the semantics of adding a `nullptr` here? That we did not find a function?

Callers that work on the result vector need to be aware that it may contain `nullptrs`.

Reply:
The semantics is that the index into the vtable at the call site is consistent with the vector of vtable entries that we are building here. Also at this time the callers are aware of the possible nullptrs.